### ISSUE TITLE
Speed up tiny incremental checks

### ISSUE THREAD

**Issue by BenWiederhake** [BenWiederhake/restic-slow#1]
tl;dr: `check` always reads *all* trees from *all* snapshots, which can take surprisingly long.

I'm using restic in such a way to invoke `backup` and then `check --read-data-subset=.1%` every 20 minutes. After I set this up, a single such run took only a minute or so. That meant I always had fresh backups of my entire home directory, *and* I could be certain that any problem would become apparent very quickly. Woohoo! :D

However, `restic check` always reads *all* trees from *all* snapshots. By now this makes up around 8 minutes of a 10 minute run, which is not so nice. Note that `rebuild-index` does not change anything, as expected.

I can think of the following workarounds, none of which are nice:
- I could continue as-is, and accept that my computer is executing `restic-check` forever.
- I make calendar entries and run `forget && prune` every month, when I'm at home and know that it can run safely for a long time. However, my hope is to have a backup system that I only have to revisit once a year, if at all. Also, "delete your backups" feels like a silly solution for a backup system. (But yes, it *does* work, that's what I've been doing a few times for various reasons.)
- I could run it like `check --with-cache`, but that basically defeats the entire purpose of `--read-data-subset`.

Here are some ideas:
- Option 1: Let's add a flag `--with-cache-but-only-for-trees-and-not-for-blobs` (better names welcome), such that the phases `load indexes`, `check all packs`, and `check snapshots, trees and blobs` become much faster (by loading everything from the local cache), and the phase `read 0.1% of data packs` still reads from the actual repo, thus making sure that the repository is still intact with high probability.
- Option 2: Let's add a flag so that `check --no-verify-trees` skips the phase `check snapshots, trees and blobs` entirely. It could still load the index(es) from the indicated destination (either local cache or the actual repository), which should be enough to sample uniformly from the blobs.
- or something else entirely, of course.

What do you think?
